ledger_secure_sdk_sys/src/lib.rs

```diff
@@ -33,4 +35,41 @@ pub fn pic_rs_mut<T>(x: &mut T) -> &mut T {
     unsafe { &mut *ptr }
 }
 
+#[cfg(feature = "heap")]
+use critical_section::RawRestoreState;
+#[cfg(feature = "heap")]
+use embedded_alloc::Heap;
+
+#[cfg(feature = "heap")]
+#[global_allocator]
+static HEAP: Heap = Heap::empty();
+
+#[cfg(feature = "heap")]
+struct CriticalSection;
+#[cfg(feature = "heap")]
+critical_section::set_impl!(CriticalSection);
+
+/// Default empty implementation as we don't have concurrency.
+#[cfg(feature = "heap")]
+unsafe impl critical_section::Impl for CriticalSection {
+    unsafe fn acquire() -> RawRestoreState {}
+    unsafe fn release(restore_state: RawRestoreState) {}
+}
+
+/// Initializes the heap memory for the global allocator.
+///
+/// The heap is stored in the stack, and has a fixed size.
+/// This method is called just before [sample_main].
+#[no_mangle]
+#[cfg(feature = "heap")]
+extern "C" fn heap_init() {
+    const HEAP_SIZE: usize = 8192;
```

Comment:
Why not use an env variable with a default value defined in the Cargo.toml of the SDK : https://doc.rust-lang.org/cargo/reference/config.html#env ?

It could be overwritten at compile time by an "advanced" user : `HEAP_SIZE=4096 cargo ledger build stax` and it would be parsed in the the lib (I know `env!` outputs a `str` but maybe there's a workaround we can implement) 

@yogh333 @kingofpayne

Reply:
Yes this is probably the best way to go. It requires converting the str to an int as a const fn, which is a bit hacky but nonetheless possible.
Also the `HEAP_SIZE` environment variable should better be defined in the `.cargo/config.toml` file.

Reply:
@kingofpayne Is it possible for you to test this solution before we merge the PR ? It would be nice to have this option from the get go.

Reply:
There's a solution I've found on stackoverflow (https://stackoverflow.com/questions/37526598/how-can-i-override-a-constant-via-a-compiler-option) for the conversion from str to int but I'm not sure this is how we want to do it as it requires another dependency on the `lazy_static` crate :

```
const MAX_DIMENSIONS_RAW: &'static str = env!("MAX_DIMENSIONS");
lazy_static! {
    static ref MAX_DIMENSIONS: usize = MAX_DIMENSIONS_RAW.parse().unwrap();
}
```

Reply:
Let's deal with this in a separate PR. I created an issue in our internal Rust SDK Jira board : https://ledgerhq.atlassian.net/browse/EMBSDK-50